src/libsyntax/parse/parser.rs

```diff
@@ -3618,7 +3618,14 @@ impl<'a> Parser<'a> {
     {
         let (iattrs, body) = self.parse_inner_attrs_and_block()?;
         attrs.extend(iattrs);
-        Ok(self.mk_expr(span_lo.to(body.span), ExprKind::TryBlock(body), attrs))
+        if self.eat_keyword(keywords::Catch) {
```

Comment:
It would be preferable to ban any form of `try { ... } $ident { .. }` from a perspective of giving up maximum flexibility in the future.

Reply:
Should those all suggest `match` instead?

Reply:
cc @estebank ^-- I would probably say yes, but that's more a diagnostics issue than future lang design flexibility.

Reply:
I'm thinking of the following case:

```
    let _ = try {
        true
    }
    let _ = ();
```

where the error is the lack of `;`, which is why I'm a bit weary of suggesting the right syntax for every ident. The current output for this the following, which I feel doesn't hinder understanding more than complaining about an inexistent `catch` would:

```
error: expected one of `.`, `;`, `?`, or an operator, found `let`
 --> src/main.rs:7:5
  |
6 |     }
  |      - expected one of `.`, `;`, `?`, or an operator here
7 |     let _ = ();
  |     ^^^ unexpected token
```